planning/behavior_path_planner/include/behavior_path_planner/scene_module/lane_change/base_class.hpp

```diff
@@ -110,6 +114,54 @@ class LaneChangeBase
     return isAbortState() ? *abort_path_ : status_.lane_change_path;
   }
 
+  PathWithLaneId getExtendPath(const PathWithLaneId & path) const
+  {
+    const auto & p = planner_data_->parameters;
+
+    const auto shift_intervals =
+      getRouteHandler()->getLateralIntervalsToPreferredLane(status_.lane_change_lanes.back());
+    const double lane_change_buffer = utils::calcMinimumLaneChangeLength(p, shift_intervals);
+
+    const auto to_lane_change_path_end =
+      calcSignedArcLength(path.points, getEgoPose().position, path.points.size() - 1);
+    const auto additional_length =
+      std::max(0.0, p.forward_path_length - to_lane_change_path_end - lane_change_buffer);
+
+    const auto original_path_end_pose = getPose(path.points.back());
+
+    constexpr double BACKWARD_MARGIN = 10.0;
+    const auto reference_lanes = getRouteHandler()->getLaneletSequence(
+      status_.lane_change_lanes.back(), original_path_end_pose, BACKWARD_MARGIN, additional_length);
+
+    const auto additional_path = utils::getCenterLinePath(
+      *getRouteHandler(), reference_lanes, original_path_end_pose, BACKWARD_MARGIN,
+      additional_length, p);
+
+    const size_t nearest_segment_idx = findFirstNearestSegmentIndexWithSoftConstraints(
+      additional_path.points, original_path_end_pose, p.ego_nearest_dist_threshold,
+      p.ego_nearest_yaw_threshold);
+
+    PathWithLaneId extended_path{};
+    {
+      extended_path.points.insert(
+        extended_path.points.end(), path.points.begin(), path.points.end());
+    }
+
+    {
+      extended_path.points.insert(
+        extended_path.points.end(), additional_path.points.begin() + nearest_segment_idx,
+        additional_path.points.end());
+    }
+
+    const size_t current_seg_idx = planner_data_->findEgoSegmentIndex(extended_path.points);
+    extended_path.points = motion_utils::cropPoints(
+      extended_path.points, getEgoPosition(), current_seg_idx, p.forward_path_length,
+      p.backward_path_length + p.input_path_interval);
+
+    constexpr double RESAMPLE_INTERVAL = 4.0;
+    return utils::resamplePathWithSpline(extended_path, RESAMPLE_INTERVAL);
```

Comment:
I not sure whether we should resample the extended path. May I know what is the reason for resampling?